coding/transform.go

```diff
@@ -5,26 +5,31 @@ import (
 
 	mc "github.com/jbenet/go-multicodec"
 	mcjson "github.com/jbenet/go-multicodec/json"
+	mccbor "github.com/jbenet/go-multicodec/cbor"
 	ipld "github.com/ipfs/go-ipld"
 )
 
-type codec struct {
+type transformCodec struct {
 	mc.Multicodec
 }
 
-type decoder struct {
+type transformDecoder struct {
 	mc.Decoder
 }
 
-func jsonMulticodec() mc.Multicodec {
-	return &codec{mcjson.Multicodec(false)}
+func JsonMulticodec() mc.Multicodec {
+	return &transformCodec{mcjson.Multicodec(false)}
 }
 
-func (c *codec) Decoder(r io.Reader) mc.Decoder {
-	return &decoder{ c.Multicodec.Decoder(r) }
+func CborMulticodec() mc.Multicodec {
+	return &transformCodec{mccbor.Multicodec()}
```

Comment:
@mildred yeah i'd want it too. maybe can have both implementations in the go-ipld package (maybe as subpackages, with one being the default or whatever). 

would be nice to have some big real workloads for go-ipfs that we could benchmark with the two interfaces.